oeps/best-practices/oep-0067-bp-tools-and-technology.rst

```diff
@@ -245,6 +245,16 @@ Frontend Technology Selection
    don't unintentionally increase the size of our JavaScript bundles, we utilize BundleWatch
    for automated bundle size monitoring.
 
+#. **Development dependencies should be separated into devDependencies**
+
+   **Rationale**: To keep installation and deployment of our MFEs as fast as
+   possible, the "dependencies" field in ``package.json`` should reference only
+   the packages needed to build the bundle of the MFE (this includes webpack via
+   frontend-build, as well as any dependencies used in the actual MFE/React
+   code). Any dependencies used only for testing, linting, formatting, or other
+   development tasks should be put into "devDependencies" (e.g. Jest, eslint,
+   TypeScript, ``@types/`` packages, etc.).
```

Comment:
I don't know how typescript is configured here, but I've had it throw errors at build time if types aren't available. Would that be a problem here?

Reply:
Good point, and I'll have to check how we're doing the build. But normally to keep builds fast, the type checking is done using `tsc` during the same time when we run the test suite / linter / code formatter / etc., and the build process uses a much faster process to merely strip out the TypeScript annotations without actually parsing them. After all, you don't need to run the linter nor the test suite during the build so neither do you need to type check.